File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {

        int s = 0;
        int e = nums.size() -1;
        int mid =0;
        int ans = -1;
        

        while(s<=e){
            mid = s + (e-s)/2;

            if(nums[mid] == target ){
                ans = mid;
                return ans;
            } else if( nums[mid] < target ) {
                ans = mid + 1;
                s = mid + 1;
                
            } else {
                e = mid-1;
                
            }
        }

        if(ans == -1) {
            return 0;
        }

        return ans;
        
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int floor = Integer.MIN_VALUE;
        int fIndex = -1;

        int s =0;
        int e = nums.length -1;
        boolean flag = false;

        while(s<=e){

            int mid = s + (e-s)/2;

            if(nums[mid] == target ) return mid;

            if(nums[mid] < target ){
                if(nums[mid] >= floor ){

                flag = true;
                fIndex = Math.max(mid,fIndex);
                floor = nums[mid];
                }
                s = mid +1;
            } else {
                e = mid -1;
            }
        }
      System.out.println("FLLOR" + floor);
        return flag ? fIndex + 1 : 0;
    }
}